src/main/java/io/debezium/connector/spanner/db/metadata/SchemaRegistry.java

```diff
@@ -66,15 +67,28 @@ public synchronized TableSchema getWatchedTable(TableId tableId) {
 
     public synchronized void checkSchema(TableId tableId, Timestamp timestamp, List<Column> rowType) {
         if (!validate(tableId, rowType)) {
-            if (this.updateSchema(timestamp)) {
-                schemaResetTrigger.run();
-                return;
+            try {
+                if (this.updateSchema(timestamp)) {
+                    schemaResetTrigger.run();
+                    updateSchemaFromStaleTimestamp(tableId, timestamp, rowType);
```

Comment:
Can we remove the call to "updateSchemaFromStaleTimestamp" here? I think it is called below to catch the exception.

Reply:
Yeah thanks for catching is. This is from a previous manual testing. Removed it.